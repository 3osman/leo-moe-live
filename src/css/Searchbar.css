.mini .Searchbar {
  padding: 0px;
}
.Searchbar {
  padding: 0px;
  display: flex;
  flex-direction: row-reverse;
  justify-content: right;
  align-items: center;
  border-radius: 50px;
}
.search-text {
  flex-grow: 2;
  font-size: 1.2rem !important;
  margin: 0 !important;
  line-height: 2.5rem;
  margin-left: 20px !important;
  height: auto !important;
  border-bottom: none !important;
  width: auto !important;
}
.search-text:focus {
  color: #000 !important;
  box-shadow: none !important;
}
.search-button {
  flex-shrink: 0;
}

a {
  line-height: 1.5;
}
