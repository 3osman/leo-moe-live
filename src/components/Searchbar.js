import React, { Component } from 'react';
import {CardPanel, Button} from 'react-materialize';
import {browserHistory} from 'react-router';
import {Link} from 'react-router'
import './../css/Searchbar.css';

class Searchbar extends Component {
  constructor(props) {
    super(props);
    this.state = {value: ((this.props.value) ? this.props.value : "" ),
                  services: {
                    youtube: this.props.youtube,
                    twitch: this.props.twitch,
                    periscope: this.props.periscope
                  }
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.search = this.search.bind(this);
    this.setSearchService = this.setSearchService.bind(this);
  }
  handleChange(event) {
    this.setState({value: event.target.value});
  }
  handleKeyPress(event) {
    if(event.key === 'Enter'){
      this.search(event);
    }
  }
  search(event) {
    event.preventDefault();
    if (this.state.value !== "" &&
        !(this.state.services.youtube === false &&
          this.state.services.twitch === false &&
            this.state.services.periscope === false)) {
      browserHistory.push('/search?value=' +
                          this.state.value +
                          '&yt=' +
                          this.state.services.youtube +
                          '&t=' +
                          this.state.services.twitch +
                          '&p=' +
                          this.state.services.periscope);
      if (this.props.mini) {
        this.props.callbackParent(this.state.value,
                                  this.state.services.youtube,
                                  this.state.services.twitch,
                                  this.state.services.persicope);
      }
    }
  }
  setSearchService(type, value) {
    if (type === 'youtube') {
      this.setState({ services: { ...this.state.services, youtube: value } });
    }
    else if (type === 'twitch') {
      this.setState({ services: { ...this.state.services, twitch: value } });
    }
    else if (type === 'periscope') {
      this.setState({ services: { ...this.state.services, periscope: value } });
    }
  }
  render() {
    return (
      <CardPanel className="Searchbar">
        <Link onClick={this.search}>
        <Button floating large
                className='red lighten-2 search-button'
                waves='light' icon='search'
                disabled={this.state.value === "" ||
                          (this.state.services.youtube === false &&
                          this.state.services.twitch === false &&
                          this.state.services.periscope === false)} />
        </Link>
        <input className="search-text grey-text grey-darken-4"
               type="text"
               name="search"
               placeholder={(this.state.value !== "") ? this.state.value : "Search live stream.."}
               value={this.state.value}
               onChange={this.handleChange}
               onKeyPress={this.handleKeyPress} />
      </CardPanel>
    );
  }
}

export default Searchbar;
