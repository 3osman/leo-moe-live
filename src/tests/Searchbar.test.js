import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, render, mount } from 'enzyme'
import Searchbar from './../components/Searchbar'
import renderer from 'react-test-renderer'

describe('searchbar', () => {
  var searchBar
  beforeEach(() => {
    searchBar = shallow(
      <Searchbar youtube="true" twitch="true" periscope="true"/>
    )
  })
  it('renders correctly', () => {
    expect(searchBar).toMatchSnapshot()
  })
})
