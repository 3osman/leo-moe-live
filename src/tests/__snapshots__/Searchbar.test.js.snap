exports[`searchbar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CardPanel
    className="Searchbar">
    <Link
        onClick={[Function]}
        onlyActiveOnIndex={false}
        style={Object {}}>
        <Button
            className="red lighten-2 search-button"
            disabled={true}
            floating={true}
            icon="search"
            large={true}
            node="button"
            waves="light" />
    </Link>
    <input
        className="search-text grey-text grey-darken-4"
        name="search"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search live stream.."
        type="text"
        value="" />
</CardPanel>,
  "nodes": Array [
    <CardPanel
      className="Searchbar">
      <Link
            onClick={[Function]}
            onlyActiveOnIndex={false}
            style={Object {}}>
            <Button
                  className="red lighten-2 search-button"
                  disabled={true}
                  floating={true}
                  icon="search"
                  large={true}
                  node="button"
                  waves="light" />
      </Link>
      <input
            className="search-text grey-text grey-darken-4"
            name="search"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search live stream.."
            type="text"
            value="" />
</CardPanel>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Searchbar
        periscope="true"
        twitch="true"
        youtube="true" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Searchbar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleKeyPress": [Function],
        "props": Object {
          "periscope": "true",
          "twitch": "true",
          "youtube": "true",
        },
        "refs": Object {},
        "search": [Function],
        "setSearchService": [Function],
        "state": Object {
          "periscope": "true",
          "twitch": "true",
          "value": "",
          "youtube": "true",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CardPanel
          className="Searchbar">
          <Link
                    onClick={[Function]}
                    onlyActiveOnIndex={false}
                    style={Object {}}>
                    <Button
                              className="red lighten-2 search-button"
                              disabled={true}
                              floating={true}
                              icon="search"
                              large={true}
                              node="button"
                              waves="light" />
          </Link>
          <input
                    className="search-text grey-text grey-darken-4"
                    name="search"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Search live stream.."
                    type="text"
                    value="" />
</CardPanel>,
        "_debugID": 2,
        "_renderedOutput": <CardPanel
          className="Searchbar">
          <Link
                    onClick={[Function]}
                    onlyActiveOnIndex={false}
                    style={Object {}}>
                    <Button
                              className="red lighten-2 search-button"
                              disabled={true}
                              floating={true}
                              icon="search"
                              large={true}
                              node="button"
                              waves="light" />
          </Link>
          <input
                    className="search-text grey-text grey-darken-4"
                    name="search"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Search live stream.."
                    type="text"
                    value="" />
</CardPanel>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Searchbar
    periscope="true"
    twitch="true"
    youtube="true" />,
}
`;
